lib/src/io_client.dart

```diff
@@ -72,3 +73,16 @@ class IOClient extends BaseClient {
     _inner = null;
   }
 }
+
+Uri _responseUrl(Request request, HttpClientResponse response) {
+  var redirects = response.redirects;
+
+  if (redirects.isEmpty) {
+    return request.url;
+  }
+
+  var location = redirects.last.location;
```

Comment:
@nex3 did not know about `p.url.joinAll()` is that still valid with the updated implementation?